subprojects/common/graph/src/main/kotlin/com/avito/graph/ShortestPath.kt

```diff
@@ -0,0 +1,174 @@
+package com.avito.graph
+
+import org.jgrapht.Graph
+import org.jgrapht.GraphPath
+import org.jgrapht.alg.shortestpath.BellmanFordShortestPath
+import org.jgrapht.graph.DefaultWeightedEdge
+import org.jgrapht.graph.DirectedWeightedMultigraph
+
+/**
+ * @operations nodes describing a DAG
+ */
+class ShortestPath<T : Operation>(private val operations: Set<T>) {
+
+    private val operationByKey: Map<String, Operation> = operations.map { it.id to it }.toMap()
+
+    fun find(): List<T> {
```

Comment:
I expect to see a more linked structure. It's not obvious that we return the path of `Operations` ordered by index
We could wrap the result with some `Path` class

Reply:
> we return the path of Operations ordered by index

This is by definition a path from a source to a sink in DAG. 
This path can't be ordered by any other reasons or attributes.

Reply:
> We could wrap the result with some Path class

Good idea. Will do.

Originally didn't want to expose `org.jgrapht.GraphPath` to hide from client implementation details with the whole graph.

Reply:
Wrapped in model in https://github.com/avito-tech/avito-android/pull/985/commits/2afbb9a4146c27e65054c5d19f28c9e37580e4cb